### PR TITLE
[WEB-1645] Code block examples are collapsible

### PR DESCRIPTION
**Pull request by bgdeutsch**
### What does this PR do?
- Makes `code-block` code examples collapsible
- NOTE:  this does not affect the following code examples:
  - generated API code examples.
  - examples using `programming-lang` shortcodes: https://docs-staging.datadoghq.com/brian.deutsch/collapsible-code-blocks/agent/docker/apm/?tab=linux#tracing-from-other-containers
  - examples dropped into markdown files using three backticks i.e ```shell here: https://raw.githubusercontent.com/DataDog/documentation/master/content/en/agent/docker/apm.md

### Motivation
- https://datadoghq.atlassian.net/browse/WEB-1645
- This is a pre-cursor to the upcoming Agent config code example updates.   We noticed those code examples could be very long and having the ability to collapse them might be useful to the user.  For example: https://docs-staging.datadoghq.com/brian.deutsch/dd-agent-config/agent/basic_agent_usage/aix/#running-dogstatsd

### Preview
- https://docs-staging.datadoghq.com/brian.deutsch/collapsible-code-blocks/agent/basic_agent_usage/aix/
- https://docs-staging.datadoghq.com/brian.deutsch/collapsible-code-blocks/agent/amazon_ecs/apm/?tab=ec2metadataendpoint

- [ ] Code blocks hide and show appropriately when clicking arrow icons
- [ ] Copy code button no longer becomes visible when hovering over the purple header area, instead only becoming visible when user hovers over the code block.   Copy code function still works.
- [ ] The functionality looks good and works at all screen sizes.
- [ ] Event listeners are re-bound by `async-loading.js` - Navigate around the site using the left-side nav (without refreshing), collapsing code blocks should still work.
- [ ] There are no console errors.

### Additional Notes
- I added a default `Code Example` title for code blocks that didn't have a file name specified, as I thought it might look confusing to the user when the code block is collapsed without a title/filename present.   Happy to revert this change if there is any concern!

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by sarina-dd** [comment]
This looks great so far, @bgdeutsch! I tested on a few pages in EN, FR, and JA, not noticing any console errors or any issues when resizing. 👍

**Review by DavidWeid — APPROVED** [review]
Looks good to me, no errors observed 👍🏼

**Review by sarina-dd — APPROVED** [review]

File: layouts/shortcodes/code-block.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{ $filename := (.Get ("filename")) }}
+{{ $filename := (.Get ("filename")) | default "Code Example" }}
```
Comment (kayayarai) [inline layouts/shortcodes/code-block.html:1]:
```suggestion
{{ $filename := (.Get ("filename")) | default "Code example" }}
```

**Comment by kayayarai** [comment]
My feedback: This works as designed. But: Having a default "heading" ("Code Example") seems heavy for one-liner code blocks, and the collapse ability isn't needed for these. 
image

But it's nice to have the "Copy" ability on these because people often want to copy a one-line command example. So we need something in between three-ticks and full shortcode code block for these.

**Review by kayayarai — APPROVED** [review]

**Pull request merged**